Git-commit: 0e90daa742079868f3ccaaab58a0a102ac614534
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Fri, 26 Jun 2020 18:11:49 +0100
Subject: [PATCH] media: i2c: imx290: Switch set_hmax to use
 imx290_write_buffered_reg
Patch-mainline: Queued in openEuler repo, version 5.10.0-4.5.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

raspberrypi inclusion
category: feature
bugzilla: 50432

--------------------------------

imx290_set_hmax was using two independent writes to set up hmax,
when all other multi-register writes were using imx290_write_buffered_reg
which claims the group hold first.

Switch imx290_set_hmax to using imx290_write_buffered_reg too.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
Signed-off-by: Fang Yafen <yafen@iscas.ac.cn>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 drivers/media/i2c/imx290.c | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/drivers/media/i2c/imx290.c b/drivers/media/i2c/imx290.c
index 09df70849928..b824dfe29bbd 100644
--- a/drivers/media/i2c/imx290.c
+++ b/drivers/media/i2c/imx290.c
@@ -568,19 +568,12 @@ static int imx290_set_hmax(struct imx290 *imx290, u32 val)
 	u32 hmax = val + imx290->current_mode->width;
 	int ret;
 
-	ret = imx290_write_reg(imx290, IMX290_HMAX_LOW, (hmax & 0xff));
-	if (ret) {
-		dev_err(imx290->dev, "Error setting HMAX register\n");
-		return ret;
-	}
-
-	ret = imx290_write_reg(imx290, IMX290_HMAX_HIGH, ((hmax >> 8) & 0xff));
-	if (ret) {
+	ret = imx290_write_buffered_reg(imx290, IMX290_HMAX_LOW, 2,
+					hmax);
+	if (ret)
 		dev_err(imx290->dev, "Error setting HMAX register\n");
-		return ret;
-	}
 
-	return 0;
+	return ret;
 }
 
 static int imx290_set_vmax(struct imx290 *imx290, u32 val)
-- 
2.31.1

